packages/common/src/tools/queue/UniqueTaskQueue.ts

```diff
@@ -0,0 +1,39 @@
+import { Logger } from '../Logger'
+import { TaskQueue } from './TaskQueue'
+import { TaskQueueOpts } from './types'
+
+export class UniqueTaskQueue<T extends string | number> extends TaskQueue<T> {
+  private taskSet: Set<T> = new Set()
+
+  constructor(
+    executeTask: (task: T) => Promise<void>,
+    logger: Logger,
+    opts?: TaskQueueOpts<T>,
+  ) {
+    const wrappedExecute = async (task: T) => {
+      await executeTask(task)
+      this.taskSet.delete(task)
+    }
+
+    super(wrappedExecute, logger, opts)
+  }
+
+  addIfEmpty(task: T): void {
```

Comment:
Shouldnt this use `override`? 

also maybe we can turn on the compiler flag to enforce its use: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#override-and-the---noimplicitoverride-flag

Reply:
Good point, updated TS rule